### DISCUSSION TITLE
Explicit H on Pyrrole and GetAtomDegree

### DISCUSSION THREAD

**Discussion by MherMatevosyan** [rdkit/rdkit#5823] [Q&A] [answered]
Hi there,

I was experimenting with some features in RDKit and noticed that `GetDegree()` doesn't take into account the Explicit Hydrogen in pyrrole, while `GetNumExplicitHs()` does. 
![image](https://user-images.githubusercontent.com/29042139/205052160-fd1690b4-65d0-472b-9be8-86e112d53e0d.png)
Is this by design? If so, what is the logic behind it?
Thanks in advance!

**Comment by Olabisi-Aishat-Bello** [comment]
Try doing mol = Chem.AddHs(mol) to explicitly define all the hydrogens in the molecule.

**Reply by greglandrum** [reply, marked as answer]
You can get the degree including implicit Hs with GetTotalDegree()

If you want the number of implicit Hs on an atom, I would suggest using GetTotalNumHs(). GetNumExplicitHs() is there for historic reasons and the value it returns is hard to explain

**Reply by MherMatevosyan** [reply]
Thanks Greg, GetTotalDegree() and GetTotalNumHs() cover the cases I was looking for.
